virtio-devices/src/balloon.rs

```diff
@@ -138,6 +193,49 @@ impl BalloonEpollHandler {
     }
 
     fn release_memory_range(
+        &mut self,
+        memory: &GuestMemoryMmap,
+        range_base: GuestAddress,
+        range_len: usize,
+    ) -> result::Result<(), Error> {
+        let page_size: u64 = get_page_size();
+        if page_size == 1 << VIRTIO_BALLOON_PFN_SHIFT {
+            return Self::discard_memory_region(memory, range_base, range_len);
+        }
+
+        if self.pbp.addr == 0 {
+            if range_base.0 & (page_size - 1) != 0 {
+                warn!(
+                    "Can't release memory region: 0x{:x} that doesn't align with page size",
+                    range_base.0
+                );
+                return Ok(());
+            }
+            // Align to page size and set it to pbp.addr
+            self.pbp.addr = align_page_size_up(range_base.0);
+        }
+
+        if !self.pbp.pfn_match(range_base.0) {
+            // We are trying to free memory region in a different pfn with current pbp. Flush addr
+            self.pbp.addr = align_page_size_up(range_base.0);
```

Comment:
I don't think this is right. IIUC, we need to flush the bit_maps with php.reset() and set the new pbp.addr with the new range_base if it is page_size aligned, right?

Reply:
Yes, may bad.